### PR TITLE
Fix the factories loading mechanism

### PR DESCRIPTION
**Pull request by elia**
**Description**

Requiring factories, although it was the correct way of doing it back in the day, has not been well supported by FactoryBot for a number of major versions<sup>1</sup>. Some applications were even relying on `require` to cherry pick only the factories they wanted.

This PR fixes both of the problems with deprecations that clearly indicate the way forward ~~and updates the version of factory_bot_rails to the latest available~~. **UPDATE:** I'll leave the upgrade to a future PR as some stuff is broken on the latest version.

**Checklist:**
- [ ] I have followed [Pull Request guidelines](https://github.com/solidusio/solidus/blob/master/CONTRIBUTING.md#pull-request-guidelines)
- [ ] I have added a detailed description into each commit message
- [ ] I have updated Guides and README accordingly to this change (if needed)
- [ ] I have added tests to cover this change (if needed)
- [ ] I have attached screenshots to this PR for visual changes (if needed)

---

_1: The latest is 6, and it's already not the right way in 4, I didn't bother searching earlier versions_

### CODE REVIEW

**Comment by kennyadsl** [comment]
@elia this is still blocked by the things broken in the last version (as you pointed out in the PR's description), right?

**Comment by elia** [comment]
> @elia this is still blocked by the things broken in the last version (as you pointed out in the PR's description), right?

I'm just excluding the update of factory bot from this PR, which anyway is focused on doing things right with the current version.
Once this is merged I can give another try at upgrading it. 👍

**Review by jarednorman — APPROVED** [review]

**Review by aldesantis — APPROVED** [review]

**Comment by kennyadsl** [comment]
@elia can we merge/release this in 3.1 or there's a specific reason to backport this change in 2.11 and 3.0 as well?

**Pull request merged**
